File: Web Game/src/constants/BrandingConstants.js
/**
 * Branding Constants for Charity: Water
 * Official brand colors, typography, and sizing
 */

const BrandingConstants = {
    // ===== OFFICIAL BRAND COLORS =====
    
    // Primary Colors
    CHARITY_WATER_BLUE: 0x2596BE,      // #2596BE
    CHARITY_WATER_BLUE_HEX: '#2596BE',
    CHARITY_WATER_BLUE_RGB: { r: 37, g: 150, b: 190 },
    
    PRIMARY_BLUE: 0x2596BE,            // Alias for CHARITY_WATER_BLUE
    PRIMARY_BLUE_HEX: '#2596BE',
    
    CHARITY_WATER_YELLOW: 0xFFC908,    // #FFC908
    CHARITY_WATER_YELLOW_HEX: '#FFC908',
    CHARITY_WATER_YELLOW_RGB: { r: 255, g: 201, b: 8 },
    
    PRIMARY_YELLOW: 0xFFC908,          // Alias for CHARITY_WATER_YELLOW
    PRIMARY_YELLOW_HEX: '#FFC908',
    
    // Supporting Colors
    TEXT_PRIMARY: 0x333333,            // #333333
    TEXT_PRIMARY_HEX: '#333333',
    TEXT_PRIMARY_RGB: { r: 51, g: 51, b: 51 },
    
    TEXT_SECONDARY: 0x666666,          // #666666
    TEXT_SECONDARY_HEX: '#666666',
    TEXT_SECONDARY_RGB: { r: 102, g: 102, b: 102 },
    
    BACKGROUND_WHITE: 0xFFFFFF,        // #FFFFFF
    BACKGROUND_WHITE_HEX: '#FFFFFF',
    BACKGROUND_WHITE_RGB: { r: 255, g: 255, b: 255 },
    
    ACCENT_LIGHT_BLUE: 0xE8F4F8,       // #E8F4F8
    ACCENT_LIGHT_BLUE_HEX: '#E8F4F8',
    ACCENT_LIGHT_BLUE_RGB: { r: 232, g: 244, b: 248 },
    
    ACCENT_LIGHT_YELLOW: 0xFFF9E6,     // #FFF9E6
    ACCENT_LIGHT_YELLOW_HEX: '#FFF9E6',
    ACCENT_LIGHT_YELLOW_RGB: { r: 255, g: 249, b: 230 },
    
    // ===== TYPOGRAPHY =====
    FONT_PRIMARY: 'Proxima Nova',
    FONT_FAMILY: "'Proxima Nova', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif",
    
    // Font Sizes
    FONT_SIZE_HEADLINE: 48,
    FONT_SIZE_SUBHEADING: 32,
    FONT_SIZE_BUTTON: 20,
    FONT_SIZE_BODY: 16,
    FONT_SIZE_LABEL: 14,
    
    // ===== SIZING CONSTANTS =====
    BUTTON_HEIGHT: 60,
    BUTTON_CORNER_RADIUS: 8,
    UI_PADDING: 16,
    UI_SPACING: 24,
    
    // ===== GRID & GAME =====
    GRID_WIDTH: 8,
    GRID_HEIGHT: 12,
    TILE_SIZE: 120,
    GAME_WIDTH: 8 * 120,    // 960px
    GAME_HEIGHT: 12 * 120,  // 1440px
    SCREEN_WIDTH: 1080,
    SCREEN_HEIGHT: 1920,
    
    // ===== HELPER METHODS =====
    
    /**
     * Get color by name
     */
    getColorByName: function(name) {
        const colors = {
            'blue': this.CHARITY_WATER_BLUE,
            'yellow': this.CHARITY_WATER_YELLOW,
            'text': this.TEXT_PRIMARY,
            'secondary': this.TEXT_SECONDARY,
            'background': this.BACKGROUND_WHITE,
            'light-blue': this.ACCENT_LIGHT_BLUE,
            'light-yellow': this.ACCENT_LIGHT_YELLOW,
        };
        return colors[name.toLowerCase()] || this.TEXT_PRIMARY;
    },
    
    /**
     * Get hex color by name
     */
    getColorHex: function(name) {
        const colors = {
            'blue': this.CHARITY_WATER_BLUE_HEX,
            'yellow': this.CHARITY_WATER_YELLOW_HEX,
            'text': this.TEXT_PRIMARY_HEX,
            'secondary': this.TEXT_SECONDARY_HEX,
            'background': this.BACKGROUND_WHITE_HEX,
            'light-blue': this.ACCENT_LIGHT_BLUE_HEX,
            'light-yellow': this.ACCENT_LIGHT_YELLOW_HEX,
        };
        return colors[name.toLowerCase()] || this.TEXT_PRIMARY_HEX;
    },
    
    /**
     * Get RGB color by name
     */
    getColorRGB: function(name) {
        const colors = {
            'blue': this.CHARITY_WATER_BLUE_RGB,
            'yellow': this.CHARITY_WATER_YELLOW_RGB,
            'text': this.TEXT_PRIMARY_RGB,
            'secondary': this.TEXT_SECONDARY_RGB,
            'background': this.BACKGROUND_WHITE_RGB,
            'light-blue': this.ACCENT_LIGHT_BLUE_RGB,
            'light-yellow': this.ACCENT_LIGHT_YELLOW_RGB,
        };
        return colors[name.toLowerCase()] || this.TEXT_PRIMARY_RGB;
    },
};

// Freeze constants to prevent accidental modification
Object.freeze(BrandingConstants);

File: Unity Project/Assets/Scripts/BrandingConstants.cs
using UnityEngine;

/// <summary>
/// Official Charity: Water Branding Constants
/// Contains all official brand colors and typography specifications
/// 
/// Brand Guide Reference: cw_Brand-UsageGuide.pdf
/// Last Updated: February 23, 2026
/// </summary>
public static class BrandingConstants
{
    // ===== PRIMARY BRAND COLORS =====
    
    /// <summary>
    /// Charity: Water Primary Blue
    /// Hex: #2596BE | RGB: (37, 150, 190)
    /// Usage: Primary color, accents, links
    /// </summary>
    public static readonly Color CHARITY_WATER_BLUE = new Color(37f/255f, 150f/255f, 190f/255f);
    
    /// <summary>
    /// Charity: Water Secondary Yellow
    /// Hex: #FFC908 | RGB: (255, 201, 8)
    /// Usage: Buttons, highlights, CTAs, call-to-action elements
    /// </summary>
    public static readonly Color CHARITY_WATER_YELLOW = new Color(255f/255f, 201f/255f, 8f/255f);
    
    // ===== SUPPORTING COLORS =====
    
    /// <summary>
    /// Primary text color - Dark Gray
    /// Hex: #333333 | RGB: (51, 51, 51)
    /// Usage: Body text, headlines, primary content
    /// </summary>
    public static readonly Color TEXT_PRIMARY = new Color(51f/255f, 51f/255f, 51f/255f);
    
    /// <summary>
    /// Secondary text color - Medium Gray
    /// Hex: #666666 | RGB: (102, 102, 102)
    /// Usage: Secondary text, subtext, muted content
    /// </summary>
    public static readonly Color TEXT_SECONDARY = new Color(102f/255f, 102f/255f, 102f/255f);
    
    /// <summary>
    /// Background color - White
    /// Hex: #FFFFFF | RGB: (255, 255, 255)
    /// Usage: Main background, panels, containers
    /// </summary>
    public static readonly Color BACKGROUND_WHITE = Color.white;
    
    /// <summary>
    /// Accent light blue - for highlights and backgrounds
    /// Hex: #E8F4F8 | RGB: (232, 244, 248)
    /// Usage: Hover states, light backgrounds, accents
    /// </summary>
    public static readonly Color ACCENT_LIGHT_BLUE = new Color(232f/255f, 244f/255f, 248f/255f);
    
    /// <summary>
    /// Accent light yellow - for highlights and backgrounds
    /// Hex: #FFF9E6 | RGB: (255, 249, 230)
    /// Usage: Hover states, light backgrounds, accents
    /// </summary>
    public static readonly Color ACCENT_LIGHT_YELLOW = new Color(255f/255f, 249f/255f, 230f/255f);
    
    // ===== TRANSPARENCY VARIANTS =====
    
    /// <summary>
    /// Charity: Water Blue with 50% alpha for overlays
    /// </summary>
    public static readonly Color CHARITY_WATER_BLUE_TRANSPARENT = new Color(37f/255f, 150f/255f, 190f/255f, 0.5f);
    
    /// <summary>
    /// Charity: Water Yellow with 50% alpha for overlays
    /// </summary>
    public static readonly Color CHARITY_WATER_YELLOW_TRANSPARENT = new Color(255f/255f, 201f/255f, 8f/255f, 0.5f);
    
    // ===== TYPOGRAPHY CONSTANTS =====
    
    /// <summary>
    /// Primary font family: Proxima Nova
    /// Make sure to import the asset into Assets/Fonts/
    /// </summary>
    public const string FONT_PROXIMA_NOVA = "Proxima Nova";
    
    // ===== FONT SIZES =====
    
    /// <summary>
    /// Font size for main headlines
    /// Usage: Scene titles, main headings
    /// Size: 48pt
    /// </summary>
    public const int FONT_SIZE_HEADLINE = 48;
    
    /// <summary>
    /// Font size for subheadings
    /// Usage: Section titles, emphasis text
    /// Size: 32pt
    /// </summary>
    public const int FONT_SIZE_SUBHEADING = 32;
    
    /// <summary>
    /// Font size for UI buttons
    /// Usage: Game buttons, calls-to-action
    /// Size: 20pt
    /// </summary>
    public const int FONT_SIZE_BUTTON = 20;
    
    /// <summary>
    /// Font size for body text
    /// Usage: Instructions, descriptions, regular content
    /// Size: 16pt
    /// </summary>
    public const int FONT_SIZE_BODY = 16;
    
    /// <summary>
    /// Font size for UI labels and small text
    /// Usage: Labels, captions, secondary copy
    /// Size: 14pt
    /// </summary>
    public const int FONT_SIZE_LABEL = 14;
    
    // ===== HEX COLOR CODES (FOR REFERENCE) =====
    
    /*
     CHARITY WATER OFFICIAL HEX CODES:
     ================================
     Primary Blue:      #2596BE
     Secondary Yellow:  #FFC908
     Text Primary:      #333333
     Text Secondary:    #666666
     Background White:  #FFFFFF
     Accent Light Blue: #E8F4F8
     Accent Light Yel:  #FFF9E6
    */
    
    // ===== SIZING CONSTANTS =====
    
    /// <summary>
    /// Standard button height
    /// </summary>
    public const float BUTTON_HEIGHT = 60f;
    
    /// <summary>
    /// Standard button corner radius
    /// </summary>
    public const float BUTTON_CORNER_RADIUS = 8f;
    
    /// <summary>
    /// Standard padding/margin for UI elements
    /// </summary>
    public const float UI_PADDING = 16f;
    
    /// <summary>
    /// Spacing between elements
    /// </summary>
    public const float UI_SPACING = 24f;
    
    // ===== HELPER METHODS =====
    
    /// <summary>
    /// Get a color by name (for editor/debugging)
    /// </summary>
    public static Color GetColorByName(string colorName)
    {
        return colorName.ToLower() switch
        {
            "blue" => CHARITY_WATER_BLUE,
            "yellow" => CHARITY_WATER_YELLOW,
            "text" => TEXT_PRIMARY,
            "secondary" => TEXT_SECONDARY,
            "background" => BACKGROUND_WHITE,
            _ => TEXT_PRIMARY
        };
    }
    
    /// <summary>
    /// Returns the Proxima Nova font (requires import)
    /// Falls back to default if not found
    /// </summary>
    public static Font GetProximaNovaFont()
    {
        Font font = Resources.Load<Font>("Fonts/" + FONT_PROXIMA_NOVA);
        return font != null ? font : Resources.GetBuiltinResource<Font>("Arial.ttf");
    }
}
